site/docs/spark.md

```diff
@@ -258,6 +258,47 @@ ALTER TABLE prod.db.sample DROP COLUMN id
 ALTER TABLE prod.db.sample DROP COLUMN point.z
 ```
 
+### `ALTER TABLE ... ADD PARTITION FIELD`
+
+```sql
+ALTER TABLE prod.db.sample ADD PARTITION FIELD catalog -- identity transform
+ALTER TABLE prod.db.sample ADD PARTITION FIELD bucket(16, id)
+ALTER TABLE prod.db.sample ADD PARTITION FIELD truncate(data, 4)
+ALTER TABLE prod.db.sample ADD PARTITION FIELD years(ts)
+-- use optional AS keyword to specify a custom name for the partition field 
+ALTER TABLE prod.db.sample ADD PARTITION FIELD bucket(16, id) AS shard
+```
+
+!!! Warning
+    Changing partitioning will change the behavior of dynamic writes, which overwrite any partition that is written to. 
```

Comment:
Based on my understanding: "dynamic writes" -> "`INSERT OVERWRITE` with dynamic overwrites"? Also "that is written to" is a bit hard to understand, but I couldn't come up with a good suggestion on this...

Reply:
Yes. Yeah the term is a bit convoluted, but I don't have a better way either. I think the example next line should make this sentence clear to readers.